### PR TITLE
#1 in TODO List

### PR DESCRIPTION
**Pull request by 50-sys**
Added the command line option --geometry for specifying the initial size of the window which is the first matter of the to-do list. Used argparse library to make it easier for adding new options in the future.

### CODE REVIEW

File: MarkupViewer.py [hunk 1]
```diff
@@ -850,15 +847,45 @@ def __init__(self, parent=None):
             for b in (spacer, self.download, self.go_to_gh, self.cancel):
                 tb.addWidget(b)
 
-            self.addToolBar(0x8, tb)
+            self.addToolBar(0x8, tb)
```
Comment (vovkkk) [inline MarkupViewer.py:850]: Trailing whitespace?

File: MarkupViewer.py [hunk 2]
```diff
@@ -850,15 +847,45 @@ def __init__(self, parent=None):
...
+            values = value.split("x")
+            setattr(namespace, self.dest, (int(values[0]), int(values[1])))
+        except:
+            error("Invalid option " + value + " for --geometry!")
+
+
+class FmAction(argparse.Action):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+
+        super().__init__(option_strings, dest, **kwargs)
```
Comment (vovkkk) [inline MarkupViewer.py:874]: Same as above.

Also it is complaining about `description` argument, I’m not sure why
```
Traceback (most recent call last):
  File ".\test.py", line 898, in <module>
    main()
  File ".\test.py", line 886, in main
    parser.add_argument('infile', nargs='?', default='', action=FmAction, description="Name of the markup file to be previewed.")
  File "C:\Python27\lib\argparse.py", line 1294, in add_argument
    action = action_class(**kwargs)
    super(FmAction, self).__init__(option_strings, dest, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'description'
```

File: MarkupViewer.py [hunk 3]
```diff
@@ -850,15 +847,45 @@ def __init__(self, parent=None):
...
+            error("Too many arguments for --geometry")
+
+        super().__init__(option_strings, dest, **kwargs)
+
+    def __call__(self, parser, namespace, value, option_string=None):
+        try:
+            values = value.split("x")
+            setattr(namespace, self.dest, (int(values[0]), int(values[1])))
+        except:
+            error("Invalid option " + value + " for --geometry!")
```
Comment (vovkkk) [inline MarkupViewer.py:868]: Might as well show an example of a valid option.

Comment (50-sys) [reply MarkupViewer.py:869]: I have worked out the other issues but don't understand what you exactly meant here. And also thank you for feedback.

Comment (vovkkk) [reply MarkupViewer.py:869]: I meant some example so user could compare an invalid option and a valid one, I’ve add it in https://github.com/vovkkk/MarkupViewer/commit/5f2fbd868efa268f4b2f4407134e2d53d7e2fcf5 also fixed some syntax errors (missing `:` and `)`)

**Pull request merged**